src/cucu/formatter/rundb.py

```diff
@@ -0,0 +1,206 @@
+        self.config = config
+
+        if (
+            not CONFIG["RUN_DB_PATH"]
+            or CONFIG["WORKER_RUN_ID"] != CONFIG["WORKER_PARENT_ID"]
+        ):
+            logger.debug(
+                "Create a new worker db since this isn't the parent process"
+            )
+            # use seed unique enough for multiple cucu_runs to be combined but predictable within the same run
+            worker_id_seed = f"{CONFIG['WORKER_PARENT_ID']}_{os.getpid()}"
+            CONFIG["WORKER_RUN_ID"] = generate_short_id(worker_id_seed)
+
+            results_path = Path(CONFIG["CUCU_RESULTS_DIR"])
+            worker_run_id = CONFIG["WORKER_RUN_ID"]
+            cucu_run_id = CONFIG["CUCU_RUN_ID"]
+            CONFIG["RUN_DB_PATH"] = run_db_path = (
+                results_path / f"run_{cucu_run_id}_{worker_run_id}.db"
+            )
+
+            if not run_db_path.exists():
+                logger.debug(
+                    f"Creating new run database file: {run_db_path} for {worker_id_seed}"
+                )
+                create_database_file(run_db_path)
+                record_cucu_run()
+
+    def uri(self, uri):
+        # nothing to do, but we need to implement the method for behave
+        pass
+
+    def feature(self, feature):
+        """Called before a feature is executed."""
+        self.this_feature = feature
+        self.this_scenario = None
+        self.this_background = None
+        self.next_start_at = None
+
+        feature_run_id_seed = (
+            f"{CONFIG['WORKER_RUN_ID']}_{time.perf_counter()}"
+        )
+        feature.feature_run_id = generate_short_id(feature_run_id_seed)
+        feature.custom_data = {}
+
+        record_feature(feature)
+
+    def background(self, background):
+        # nothing to do, but we need to implement the method for behave
+        pass
+
+    def _finish_scenario(self):
+        if self.this_scenario is None:
+            return
+
+        # ensure non-executed steps have correct seq
+        for index, step in enumerate(self.steps):
+            if getattr(step, "seq", -1) == -1:
+                step.seq = index + 1  # 1-based sequence
+                finish_step_record(step, None)
+
+        finish_scenario_record(self.this_scenario)
+
+    def scenario(self, scenario):
+        """Called before a scenario is executed (or ScenarioOutline scenarios)."""
+        self._finish_scenario()
+
+        self.this_scenario = scenario
+        self.steps = []
+        self.next_start_at = datetime.datetime.now().isoformat()[:-3]
+        scenario_run_id_seed = (
+            f"{scenario.feature.feature_run_id}_{time.perf_counter()}"
+        )
+        scenario.scenario_run_id = generate_short_id(scenario_run_id_seed)
+        scenario.custom_data = {}
+
+        # search features.scenarios, which includes scenario outlines and descend into them
+        # for scenarios that are part of scenario outlines we combine the outline index and the scenario index
```

Comment:
The wording of this sentence is hard to follow

Reply:
how about this then? (somehow this comment didn't track with the update)
```
        # feature.scenarios is a mix of Scenario and ScenarioOutline objects with scenarios,
        # so also iterate through their scenarios and add `.x` suffix based on the scenario outline order
```

Reply:
If I may:
```suggestion
        scenario.custom_data = {}

        # Scenarios belonging to a Scenario Outline are included in the index of their Scenario Outline.
        # Add an index as a suffix here to preserve the correct ordering.
```
This is my best understanding of what's happening here based on the comment and the code.